src/TesApi.Web/Storage/StorageAccessProvider.cs

```diff
@@ -95,7 +95,15 @@ public async Task<string> DownloadBlobAsync(Uri blobAbsoluteUrl, CancellationTok
     /// <inheritdoc/>
     public async Task<IList<Uri>> GetBlobUrlsAsync(Uri blobVirtualDirectory, CancellationToken cancellationToken)
     {
-        return (await AzureProxy.ListBlobsAsync(blobVirtualDirectory, cancellationToken)).Select(b => b.Uri).ToList();
+        Azure.Storage.Blobs.BlobUriBuilder blobBuilder = new(blobVirtualDirectory) { Sas = null };
+        return (await AzureProxy.ListBlobsAsync(blobVirtualDirectory, cancellationToken)).Select(GetBlobUri).ToList();
+
+        Uri GetBlobUri(Azure.Storage.Blobs.Models.BlobItem blob)
+        {
+            // This implementation assumes there are no parallel invocations in the same parent method call.
```

Comment:
any concerns about thread safety or this resulting in random bugs?

Reply:
Not with the current implementation. If this is refactored in the future, there could be a concern. That's why I added this comment.

Reply:
I'll update the comment to describe the concern